apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -53,175 +44,8 @@ where
-                                                )
-                                                .into();
-                                            response
-                                                .events
-                                                .push(proposal_event);
-
-                                            treasury_address
-                                        }
-                                    }
-                                    Err(_e) => {
-                                        self.write_log.drop_tx();
-                                        let proposal_event: Event =
-                                            ProposalEvent::new(
-                                                EventType::Proposal.to_string(),
-                                                TallyResult::Passed,
-                                                id,
-                                                true,
-                                                false,
-                                            )
-                                            .into();
-                                        response.events.push(proposal_event);
-
-                                        treasury_address
-                                    }
-                                }
-                            }
-                            None => {
-                                let proposal_event: Event = ProposalEvent::new(
-                                    EventType::Proposal.to_string(),
-                                    TallyResult::Passed,
-                                    id,
-                                    false,
-                                    false,
-                                )
-                                .into();
-                                response.events.push(proposal_event);
-
-                                proposal_author
-                            }
-                        }
-                    }
-                    TallyResult::Rejected | TallyResult::Unknown => {
-                        let proposal_event: Event = ProposalEvent::new(
-                            EventType::Proposal.to_string(),
-                            TallyResult::Rejected,
-                            id,
-                            false,
-                            false,
-                        )
-                        .into();
-                        response.events.push(proposal_event);
-
-                        treasury_address
-                    }
-                };
-
-                // transfer proposal locked funds
-                self.storage.transfer(
-                    &m1t(),
-                    funds,
-                    &gov_address,
-                    &transfer_address,
-                );
-            }
-        }
+        let _proposals_result =
+            execute_governance_proposals(self, new_epoch, &mut response)?;
```

Comment:
just a little nit - I think it may be a bit nicer to check the bool here:

```
if new_epoch {
  let _proposals_result =
            execute_governance_proposals(self, &mut response)?;
}
```
rather than passing it in